Iam 160 create charmed operator (#2)
canonical/identity-platform-login-ui-operator: CONTRIBUTING.md

# Contributing

## Overview

This document explains the processes and practices recommended for contributing enhancements to
this operator.

- Generally, before developing bugs or enhancements to this charm, you should [open an issue

>   ](https://github.com/canonical/hydra-operator/issues) explaining your use case.

Reviewer: ```suggestion
  ](https://github.com/canonical/identity-platform-login-ui-operator/issues) explaining your use case.
```

Author: done

source .tox/unit/bin/activate
```

### Testing

```shell
tox -e fmt           # update your code according to linting rules
tox -e lint          # code style
tox -e unit          # unit tests
tox                  # runs 'lint' and 'unit' environments
```

## Build charm

Build the charm in this git repository using:

```shell
charmcraft pack
```

### Deploy

```bash
# Create a model
juju add-model dev
# Enable DEBUG logging
juju model-config logging-config="<root>=INFO;unit=DEBUG"
# Deploy identity-platform-login-ui charm

> juju deploy ./identity-platform-login-ui_ubuntu-*.charm --resource oci-image=$(yq eval '.resources.oci-image.upstream-source' metadata.yaml)

Reviewer: ```suggestion
juju deploy ./identity-platform-login-ui-operator_ubuntu-*.charm --resource oci-image=$(yq eval '.resources.oci-image.upstream-source' metadata.yaml)
```

Reviewer: We also need to add `--trust` to the deploy command as you're using `KubernetesServicePatch` - otherwise deployment would fail when rbac is enabled

Author: done

```

## Canonical Contributor Agreement

> Canonical welcomes contributions to Charmed Ory Hydra. Please check out our [contributor agreement](https://ubuntu.com/legal/contributors) if you're interested in contributing to the solution.

Reviewer: ```suggestion
Canonical welcomes contributions to Charmed Identity Platform Login UI. Please check out our [contributor agreement](https://ubuntu.com/legal/contributors) if you're interested in contributing to the solution.
```

Author: done
